mongo/job_store.go

```diff
@@ -239,6 +284,26 @@ func (m *JobStore) UpdateJob(updates bson.M, s *mgo.Session, id string) error {
 	return nil
 }
 
+// UpsertTask creates a new task document or overwrites an existing one
+func (m *JobStore) UpsertTask(jobID, taskName string, task models.Task) (err error) {
+	s := m.Session.Copy()
+	defer s.Close()
+
+	selector := bson.M{
+		"task_name": taskName,
+		"job_id":    jobID,
```

Comment:
Do we want to check job ID is not empty?

Reply:
The CreateTask function is the only one that calls this method. That function already checks to see if the job id is empty so it doesn't really need to be checked again here unless it's going to be called by something else - it isn't currently though.